src/reimer_arenkiel_lab_to_nwb/dj_utils.py

```diff
@@ -0,0 +1,157 @@
+from pynwb import NWBFile
+from pynwb.behavior import SpatialSeries
+from pynwb.file import Subject
+from pynwb.image import GrayscaleImage
+from scipy.interpolate import interp1d
+from pynwb.base import TimeSeries, Images
+import datajoint as dj
+
+
+dj.conn()
+
+odor = dj.create_virtual_module('odor', 'pipeline_odor')
+stimulus = dj.create_virtual_module('stimulus', 'pipeline_stimulus')
+treadmill = dj.create_virtual_module('treadmill', 'pipeline_treadmill')
+mice = dj.create_virtual_module('mice', 'common_mice')
+odor = dj.create_virtual_module('odor', 'pipeline_odor')
+meso = dj.create_virtual_module('meso', 'pipeline_meso')
+
+
+def add_treadmill(nwbfile: NWBFile, animal_id: int = 125, odor_session: int = 0):
+    """Fetch treadmill data and synchronize to odor clock using linear interpolation with extrapolation"""
+
+    restriction = odor.MesoMatch & {'animal_id':animal_id, 'odor_session': odor_session}
+
+    odor_scan_times = (odor.OdorSync & restriction).fetch1('frame_times')
+    beh_scan_times = (stimulus.BehaviorSync & restriction).fetch1('frame_times')
+    beh_tread_times, tread_vel, tread_raw = (treadmill.Treadmill & restriction).fetch1('treadmill_time', 'treadmill_vel', 'treadmill_raw')
+
+    beh_odor_interp = interp1d(beh_scan_times, odor_scan_times,  kind='linear', fill_value="extrapolate")
+    odor_tread_times = beh_odor_interp(beh_tread_times)
+
+    treadmill_raw_spatial_series = SpatialSeries(
+        name="treadmill_position",
+        description="treadmill position from Treadmill table",
+        data=tread_raw,
+        timestamps=odor_tread_times,
+        reference_frame="unknown",
+    )
+
+    nwbfile.add_acquisition(treadmill_raw_spatial_series)
+
+    if "behavior" not in nwbfile.processing:
+        nwbfile.create_processing_module(name="behavior", description="behavioral data processing")
+
+    nwbfile.processing["behavior"].add(
+        TimeSeries(
+            name="treadmill_velocity",
+            description="treadmill velocity from Treadmill table",
+            data=tread_vel,
+            timestamps=treadmill_raw_spatial_series,
+            unit="unknown",
+        )
+    )
+
+
+def add_subject(nwbfile: NWBFile, animal_id: int = 125):
+    """Fetch subject data and add to NWBFile"""
+
+    subject_info = (mice.Mice & {'animal_id':animal_id}).fetch1()
+    nwbfile.subject = Subject(
+        subject_id=str(subject_info['animal_id']),
+        date_of_birth=subject_info['dob'],
```

Comment:
Here, I got this error: 
```python
Exception has occurred: TypeError
Subject.__init__: incorrect type for 'date_of_birth' (got 'date', expected 'datetime')
  File "/home/amtra/CatalystNeuro/reimer-arenkiel-lab-to-nwb/src/reimer_arenkiel_lab_to_nwb/dj_utils.py", line 62, in add_subject
    nwbfile.subject = Subject(
                      ^^^^^^^^
  File "/home/amtra/CatalystNeuro/reimer-arenkiel-lab-to-nwb/src/reimer_arenkiel_lab_to_nwb/dj_utils.py", line 155, in <module>
    add_subject(nwbfile)
TypeError: Subject.__init__: incorrect type for 'date_of_birth' (got 'date', expected 'datetime')
```
Is it okay to change it to have the time set at midnight?

```suggestion
        date_of_birth= datetime.combine(subject_info['dob'], datetime.min.time()),
```